packages/react/src/ElementsInput/ElementsInput.utils.ts

```diff
@@ -0,0 +1,155 @@
+import { InternalSchemaElement, InternalTypeSchema, TypedValue } from '@medplum/core';
+import React from 'react';
+
+/**
+ * Splits a string on the last occurrence of the delimiter
+ * @param str - The string to split
+ * @param delim - The delimiter string
+ * @returns An array of two strings; the first consisting of the beginning of the
+ * string up to the last occurrence of the delimiter. the second is the remainder of the
+ * string after the last occurrence of the delimiter. If the delimiter is not present
+ * in the string, the first element is empty and the second is the input string.
+ */
+function splitOnceRight(str: string, delim: string): [string, string] {
+  const delimIndex = str.lastIndexOf(delim);
+  if (delimIndex === -1) {
+    return ['', str];
+  }
+  const beginning = str.substring(0, delimIndex);
+  const last = str.substring(delimIndex + delim.length);
+  return [beginning, last];
+}
+
+export type ElementsContextType = {
+  debugMode: boolean;
+  profileUrl: string | undefined;
+  /**
+   * Get the element definition for the specified path if it has been modified by a profile.
+   * @param nestedElementPath - The path of the nested element
+   * @returns The modified element definition if it has been modified by the active profile or undefined. If undefined,
+   * the element has the default definition for the given type.
+   */
+  getModifiedNestedElement: (nestedElementPath: string) => InternalSchemaElement | undefined;
+  getElementByPath: (path: string) => InternalSchemaElement | undefined;
+  elements: Record<string, InternalSchemaElement>;
+  elementsByPath: Record<string, InternalSchemaElement>;
+  fixedProperties: { [key: string]: InternalSchemaElement & { fixed: TypedValue } };
+};
+
+export const ElementsContext = React.createContext<ElementsContextType>({
+  profileUrl: undefined,
+  debugMode: false,
+  getModifiedNestedElement: () => undefined,
+  getElementByPath: () => undefined,
+  elements: Object.create(null),
+  elementsByPath: Object.create(null),
+  fixedProperties: Object.create(null),
+});
+ElementsContext.displayName = 'ElementsContext';
+
+export type BuildElementsContextArgs = {
+  elements: InternalTypeSchema['elements'] | undefined;
+  parentPath: string;
+  parentContext: ElementsContextType | undefined;
+  parentType: string | undefined;
+  profileUrl?: string;
+  debugMode?: boolean;
+};
+
+function hasFixed(element: InternalSchemaElement): element is InternalSchemaElement & { fixed: TypedValue } {
+  return Boolean(element.fixed);
```

Comment:
This check could be stricter to align with the predicate in the signature:
```suggestion
  return element.fixed?.type && Object.hasOwnProperty(element.fixed, 'value');
```